TPGSup/TPG300_switching_functions.db

```diff
@@ -0,0 +1,92 @@
+record(bo, "$(P)FUNCTION:$(Q):SIM")
+{
+    field(SCAN, "Passive")
+    field(DTYP, "Soft Channel")
+    field(ZNAM, "NO")
+    field(ONAM, "YES")
+    field(VAL, "$(RECSIM=0)")
+}
+
+record(bo, "$(P)FUNCTION:$(Q):DISABLE") 
+{
+    field(DESC, "Disable comms")
+    field(PINI, "YES")
+    field(VAL, "$(DISABLE=0)")
+    field(OMSL, "supervisory")
+    field(ZNAM, "COMMS ENABLED")
+    field(ONAM, "COMMS DISABLED")
+}
+
+record(fanout, "$(P)FUNCTION:$(Q):RB:FAN")
+{
+    field(SCAN, "Passive")
+    field(LNK1, "$(P)FUNCTION:$(Q):LOW:RB")
+    field(LNK2, "$(P)FUNCTION:$(Q):LOW:E:RB")
+    field(LNK3, "$(P)FUNCTION:$(Q):HIGH:RB")
+    field(LNK4, "$(P)FUNCTION:$(Q):HIGH:E:RB")
+    field(LNK5, "$(P)FUNCTION:$(Q):ASSIGN:RB")
+}
+
+record(scalcout, "$(P)FUNCTION:$(Q):LOW:RB")
+{
+    field(DESC, "Function $(Q) low threshold readback")
+    field(SCAN, "Passive")
+    field(INAA, "$(P)FUNCTION:$(Q):RB")
+    field(CALC, "AA[0,'E']")
+    field(PREC, "1")
+    info(INTEREST, "MEDIUM")
+}
+
+record(scalcout, "$(P)FUNCTION:$(Q):LOW:E:RB")
+{
+    field(DESC, "Function $(Q) low threshold exponent readback")
+    field(SCAN, "Passive")
+    field(INAA, "$(P)FUNCTION:$(Q):RB")
+    field(CALC, "AA['E',',']")
+    info(INTEREST, "MEDIUM")
+}
+
+record(scalcout, "$(P)FUNCTION:$(Q):HIGH:RB")
+{
+    field(DESC, "Function $(Q) high threshold readback")
+    field(SCAN, "Passive")
+    field(INAA, "$(P)FUNCTION:$(Q):RB")
+    field(CALC, "((AA[',',-1])[0,','])[0,'E']")
+    field(PREC, "1")
+    info(INTEREST, "MEDIUM")
+}
+
+record(scalcout, "$(P)FUNCTION:$(Q):HIGH:E:RB")
+{
+    field(DESC, "Function $(Q) high threshold exponent readback")
+    field(SCAN, "Passive")
+    field(INAA, "$(P)FUNCTION:$(Q):RB")
+    field(CALC, "((AA[',',-1])[0,','])['E',-1]")
```

Comment:
If there's a good reason we need to keep doing the parsing in `scalcout` records (see my comment above) then I feel this would benefit from significantly more commenting/examples of how this extracts the correct information... this is a somewhat tricky expression - even more so when you have to go back and forth between the code and the manual to get the response format.

But see my other comments - I'm not convinced this is the best approach.